### PR TITLE
Plugin client only paths

### PR DESCRIPTION
**Pull request by scottyeck**
Moving the functionality described [here](https://www.gatsbyjs.org/docs/creating-and-modifying-pages/) into a plugin. I've got a follow-up PR that uses the plugin in the _client-only-paths_ example, but we need to publish this first.

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Nifty! I'll probably bikeshed you a bit on the names 😈  but I really like the idea of a higher-level declarative interface to this.

Let's also update this page https://www.gatsbyjs.org/docs/creating-and-modifying-pages/#creating-client-only-routes

**Comment by scottyeck** [comment]
Bikeshed away. Naming is hard. Also not sure if the `*` is `/app/*` is even necessary, but it's a start.

**Comment by scottyeck** [comment]
@KyleAMathews Any thoughts on those names?

File: packages/gatsby-plugin-client-only-paths/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "gatsby-plugin-client-only-paths",
+  "version": "1.0.0",
+  "description": "Stub description for gatsby-plugin-client-only-paths",
```
Comment (KyleAMathews) [inline packages/gatsby-plugin-client-only-paths/package.json]: Update this :-)

**Comment by scottyeck** [comment]
Will rename to `gatsby-plugin-create-client-paths`

**Comment by scottyeck** [comment]
@KyleAMathews - I've renamed as discussed.

**Comment by KyleAMathews** [comment]
Thanks!

**Pull request merged**